modules/core/src/main/java/com/smartbear/readyapi4j/support/RecipeLogger.java

```diff
@@ -40,12 +43,44 @@ public void filterRecipe(TestRecipe testRecipe) {
                 directory.mkdirs();
             }
 
-            File file = File.createTempFile(prefix, "." + extension, directory);
+            File file;
+            String name = testRecipe.getName();
+            if(StringUtils.isNotBlank(name)){
+                file = new File( directory, createFileName( name, '_') + "." + extension );
+            }
+            else {
+                file = File.createTempFile(prefix, "." + extension, directory);
+            }
             try ( FileOutputStream fileOutputStream = new FileOutputStream(file)) {
-                fileOutputStream.write(testRecipe.toString().getBytes());
+                fileOutputStream.write(prettyPrintRecipe(testRecipe).getBytes());
             }
         } catch (Exception e) {
             LOG.error("Failed to write recipe to file", e);
         }
     }
+
+    private String prettyPrintRecipe(TestRecipe testRecipe) {
+        try {
+            return Json.pretty( Json.mapper().readTree(testRecipe.toString()));
+        } catch (IOException e) {
+            return testRecipe.toString();
+        }
+    }
+
+    public static String createFileName(String str, char whitespaceChar) {
+        StringBuilder result = new StringBuilder();
```

Comment:
If you accept leaving the 0 char in the string, you could replace the whole method with return str.replaceAll("\\s", String.valueOf(whitespaceChar))